README.md
# NGBTO-Latest-known-version-

Newbies go back to overkill/changelog.txt
Modification made for Payday 2 by FishTaco
---------------------- 76 ----------------------
Updated at 01-09-2017
BLT 2 Support
Fixed error printing due to json files
---------------------- 75 ----------------------
Updated at 01-09-2017
BLT 2 Support
Localization Fix provided by hollerbach
---------------------- 3.47 R74 ----------------------
Updated at 07-10-2016
Hotfix for crashing when playing as client.
---------------------- 3.46 R73 ----------------------
Updated at 07-10-2016
Added Mayhem support.
Added One Down support.
---------------------- 3.45 R72 ----------------------
Updated at 27-08-2016
Fixed crashing when you get kicked while loading.
Tweaked the mod name when using it on Overkill difficulty.
---------------------- 3.44 R71 ----------------------
Updated at 03-07-2016
Now you can kick people by any deployable.
Deleted localized message, no point for it.
Fixed system prints not appearing if you weren't the host.
Tweaked the warn for a kick to appear only when a player loads.
---------------------- 3.43 R70 ----------------------
Updated at 03-07-2016
Fixed deployable kicking not working properly.
Fixed kick message being invisible if you manually kick a person.
Changed the settings for deployable checks, it wont allow anyone to spawn with disallowed deployables(either ingame or in loadout screen).
---------------------- 3.42 R69 ----------------------
Updated at 01-07-2016
Added detailed kick list. It will print to you and the players when they join what deployables, perk decks and armors are unallowed.
Fixed the perk deck related checks so you can't kick yourself.
Fixed private profiles being kicked even if the mod was turned off.
---------------------- 3.41 R68 ----------------------
Updated at 01-07-2016
Fixed greet message being sent from other players.
Added additional check if the message was printed too fast for cheater messages.
---------------------- 3.40 R67 ----------------------
Updated at 30-06-2016
Fixed crash related to infamy checking.
---------------------- 3.39 R66 ----------------------
Updated at 30-06-2016
Fixed deployables not working correctly.
Changed deployables will now have different options to kick for no matter how many: ecm jammers, sentry guns, bodybags the player has.
---------------------- 3.38 R65 ----------------------
Updated at 30-06-2016
Fix for crash mentioned by Anonymous FuX.
Updated Dutch localization by BitsOfSkin.
---------------------- 3.37 R64 ----------------------
Updated at 30-06-2016
Redone skill fetching.
Added Second deployable checking.
---------------------- 3.36 R63 ----------------------
Updated at 29-06-2016
Added detection rate filters(one for stealth and one for loud).
Fixed toggle button.
---------------------- 3.35 R62 ----------------------
Updated at 28-06-2016
Fixed the hours for the infamy check being incorrect.
---------------------- 3.34 R61 ----------------------
Updated at 28-06-2016
Added ability to filter if the person has reached certain infamy faster than your selected value.
Added ability to filter if the person has more than 350 achievements in less than 300 hours.
---------------------- 3.33 R60 ----------------------
Updated at 28-06-2016
Fixed the toggle button turning on the mod after purchasing any heist.
Disabled the system kick message if you are the host.
Fixed the empty perk deck print.
Added an ability to annouce players why they were kicked. It will spam the message to the kicked person three times and after a certain amount of time, they will get kicked.
Added an extra option to decide how to kick people, do you want to let him rejoin or not.
Added ingame changelog annouce.
---------------------- 3.32 R59 ----------------------
Updated at 26-06-2016
Moved out the blacklist to the main game directory, will simplify the updates for JackHud and me.
Improved english localization.
Added a selection to either use skill point anti-cheat or not.
---------------------- 3.31 R58 ----------------------
Updated at 25-06-2016
Fixed Chinese and Russian languages causing instant crash.
Improved pd2stats lookup, it will work in a more civil manner.
---------------------- 3.30 R57 ----------------------
Updated at 25-06-2016
Code optimization.
Lowered maximum hour selection for overkill to 100.
Split all the options menus to submenus, you won't have to use that clunky big menu anymore.
Added Biker perk deck filter.
---------------------- 3.27 R56 ----------------------
Updated at 23-06-2016
Removed the skill sharing function, the main game update will now force everyone to share their skill points.
Removed the skull/ghost icons from print messages.
Added an ability to print either detailed skill info or just a quick summary per tree.
Fixed the max skill point count for a person until you can call him a cheater.
---------------------- 3.26 R55 ----------------------
Updated at 13-06-2016
Reenabled pd2stats.
Fixed when you as a client don't get notifications about other people skills as they join.
Added Kill per minute to statistics.
---------------------- 3.25 R54 ----------------------
Updated at 04-06-2016
Updated the anti cheat for skills to work according the levels.
Added the skill tree sharing to the mod, so now you will share your progression with others, well and you will see your own.
---------------------- 3.24 R53 ----------------------
Updated at 04-06-2016
Tweaked the mod, so skill point check would work with the new skill tree system.
---------------------- 3.23 R52 ----------------------
Updated at 03-06-2016
Force disabled pd2stats checks, everyone keeps on getting tagged as a cheater. (Will reenable when it goes back to normal)
---------------------- 3.22 R51 ----------------------
Updated at 02-06-2016
Fixed German localization.
---------------------- 3.21 R50 ----------------------
Updated at 02-06-2016
Fixed German localization.
---------------------- 3.2 R49 ----------------------
Updated at 02-06-2016

Newbies go back to overkill/lua/NetworkHooks.lua
if RequiredScript == "lib/network/base/networkpeer" then

	Hooks:PostHook(HostNetworkSession, "init", "NoobJoin:Stuff", function(self, ...)
		for k, v in ipairs(NoobJoin.blacklist or {}) do
			self._kicked_list[v] = true
		end
	end)

	Hooks:Add("NetworkManagerOnPeerAdded", "NoobJoin:PeerAdded", function(peer, peer_id)
		NoobJoin:PlayerJoin(peer_id)
	end)

	Hooks:Add("BaseNetworkSessionOnLoadComplete", "NoobJoin:LoadComplete", function(peer, id)
		NoobJoin.LoadCompleted = true
		DelayedCalls:Add("NoobJoin:Skkils_" .. tostring(id), 0.5 , function()
			if managers.network._session and #managers.network:session():peers() > 0 then
				NoobJoin:Skills(id)
				for i=1,#NoobJoin.synced do
					NoobJoin:Message_Receive(NoobJoin.synced[i][1], NoobJoin.synced[i][2])
				end
				for _, peer in pairs(managers.network:session():peers()) do
					NoobJoin:Player_Announce(peer)
				end
			end
		end)
	end)

	Hooks:Add("BaseNetworkSessionOnPeerRemoved", "NoobJoin:PeerRemoved", function(peer, peer_id, reason)
		for j=1,9 do -- Skill printed, cheater, skills for overlay, join time, hours played
			NoobJoin.Players[peer_id][j] = 0
			if NoobJoin.Kick_on_join[1] == true then
				NoobJoin.Kick_on_join[1] = false
			end
		end
	end)

	Hooks:PostHook(NetworkPeer, "mark_cheater", "NoobJoin:CheaterCaught", function(self, reason, auto_kick)
		if NoobJoin.settings.ingame_anticheat == true and self:id() ~= managers.network:session():local_peer():id() then
			local message = string.format("%s: %s",  self:name(), managers.localization:text("pd_anticheat_self"))
			NoobJoin:Message_Receive(message, 1)
			NoobJoin:Add_Cheater(self:user_id(), self:name(), message)
		end
	end)

	Hooks:PostHook(NetworkPeer, "set_outfit_string", "NoobJoin:SetOutfit", function(self, outfit_string, outfit_version, outfit_signature)
		DelayedCalls:Add("Skills" .. tostring(self:id()), 0.5 , function()
			NoobJoin:Skills(self:id())
		end)
	end)

	Hooks:PostHook(NetworkPeer, "set_loading", "NoobJoin:Loaded", function(self, state)
		if self._loaded == true then
			if Utils:IsInGameState() then
				NoobJoin:Join_Message(self:id())
			end
			if NoobJoin.Kick_on_join[self:id()] then
				if NoobJoin.Kick_on_join[self:id()][1] == true then
					NoobJoin:Slow_kick(self:id(), NoobJoin.Kick_on_join[self:id()][2], NoobJoin.Kick_on_join[self:id()][3])
				end
			end
		end
	end)
end

if RequiredScript == "lib/network/matchmaking/networkmatchmakingsteam" then
	local _is_server_ok_original = NetworkMatchMakingSTEAM.is_server_ok
	function NetworkMatchMakingSTEAM:is_server_ok(friends_only, room, ...)
		for _,line in pairs(NoobJoin.blacklist) do
			if line == room then
				return false
			end
		end
		return _is_server_ok_original(self, friends_only, room, ...)
	end
end

if RequiredScript == "lib/network/base/basenetworksession" then
	Printed = false
	Current_time = 0
	Hooks:PostHook(BaseNetworkSession, "on_set_member_ready", "NoobJoin:Deployables", function(self, peer_id, ready, state_changed, from_network)
			NoobJoin:Deployables_Lookup(peer_id)
	end)

	Hooks:PostHook(BaseNetworkSession, "on_statistics_recieved", "NoobJoin:Stats", function(self, peer_id, peer_kills, peer_specials_kills, peer_head_shots, accuracy, downs)
		if Printed == false then
			Printed = true
			Current_time = managers.game_play_central and managers.game_play_central:get_heist_timer() or 0
			DelayedCalls:Add("NoobJoin:Statistics_message", 0.5, function()
				if NoobJoin.settings.Stats_Print == true then
					local message = managers.localization:text("Newbie_Statistics") .. ":" .. " " .. managers.localization:text("Newbie_Kills") .. " | " .. managers.localization:text("Newbie_Kill_Per_Minute") .. " | " .. managers.localization:text("Newbie_Spec_Kills") .. " | " .. managers.localization:text("Newbie_Headshots") .. " | " .. managers.localization:text("Newbie_Accuracy") .. " | " .. managers.localization:text("Newbie_Downs")
					NoobJoin:Message_Receive(message, 2)
				end
			end)
		end
		DelayedCalls:Add("NoobJoin:Statistics_message_peer_" .. tostring(peer_id) , 0.7, function()
			if NoobJoin.settings.Stats_Print == true then
				local peer = managers.network:session():peer(peer_id)
				local kpm = peer_kills
				if Current_time > 60 and peer_kills > 0 then
					kpm =(math.floor(((peer_kills / ((Current_time/60)-(NoobJoin.Players[peer_id][4]/60))) + 1/2)/1) * 1)
				end
				if peer:has_statistics() then
					local message = peer:name() .. " | K:" .. peer_kills .. " | KPM:" .. kpm .. " | S/K:" .. peer_specials_kills .. " | H:" .. peer_head_shots .. " | A:" .. accuracy .. " % | D:" .. downs .. " "
					NoobJoin:Message_Receive(message, 2)
				end
			end
		end)
	end)
end

if RequiredScript == "lib/managers/hudmanagerpd2" then
	Hooks:PostHook(HUDManager, "set_slot_outfit", "NoobJoin:Detection", function(self, peer_id, criminal_name, outfit, ...)
		NoobJoin.Players[peer_id][7] = tonumber(string.format("%.0f", managers.blackmarket:get_suspicion_offset_of_outfit_string(outfit, tweak_data.player.SUSPICION_OFFSET_LERP or 0.75) * 100))
	end)
end

if RequiredScript == "lib/managers/playermanager" then
	Hooks:PostHook(PlayerManager, "set_synced_deployable_equipment", "NoobJoin:DeployablesSwitch", function(self, peer, deployable, amount)
		if Utils:IsInHeist() and NoobJoin.settings.Toggle == 1 and peer:id() ~= 1 and Network:is_server() then
			NoobJoin:Deployables(peer:id(), deployable, amount)
		end
	end)
end


Newbies go back to overkill/lua/OptionsMenu.lua
if not _G.NoobJoin then
	_G.NoobJoin = _G.NoobJoin or {}
	NoobJoin._path = ModPath
	NoobJoin._data_path = SavePath .. "newbies_go_back_to_overkill.txt"
	NoobJoin.settings = {
		kickhidden_val = false,
		min_hours_loud_dw_val = 6,
		min_hours_stealth_dw_val = 7,
		min_hours_loud_ovk_val = 4,
		min_hours_stealth_ovk_val = 6,
		min_hours_loud_sm_wish_val = 10,
		min_hours_stealth_sm_wish_val = 10,
		min_hours_loud_easy_wish_val = 7,
		min_hours_stealth_ovk_val = 8,
		Toggle = 1,
		localised_message = false,
		broadcastinfo_val = true,
		perkdeck0_val = false,
		perkdeck_1_val = false,
		perkdeck_2_val = false,
		perkdeck_3_val = false,
		perkdeck_4_val = false,
		perkdeck_5_val = false,
		perkdeck_6_val = false,
		perkdeck_7_val = false,
		perkdeck_8_val = false,
		perkdeck_9_val = false,
		perkdeck_10_val = false,
		perkdeck_11_val = false,
		perkdeck_12_val = false,
		perkdeck_13_val = false,
		perkdeck_14_val = false,
		perkdeck_15_val = false,
		perkdeck_16_val = false,
		perkdeck_17_val = false,
		perkdeck_18_val = false,
		perkdeck_19_val = false,
		perkdeck_20_val = false,
		show_perks_info_val = true,
		hiddeninfamycheck = false,
		hiddeninfamy = 2,
		hiddeninfamymax = 26,
		noequipment = false,
		singleturret = false,
		bodybags = false,
		armorkit = false,
		singleecm = false,
		fourfaks = false,
		armor1 = false,
		armor2 = false,
		armor3 = false,
		armor4 = false,
		armor5 = false,
		armor6 = false,
		armor7 = false,
		incompletedeck = false,
		hiddenxxv = false,
		hiddenv = false,
		usepd2statsanticheat = true,
		tag_not_enough_heists_completed = false,
		deathwish_count_enable = false,
		deathwish_count = 1,
		total_count_enable = false,
		total_count = 1,
		Noob_language = 1,
		Stats_Print = true,
		ingame_anticheat = true,
		skills_detailed = false,
		skill_cheats = true,
		message = 1,
		kick_type = true,
		Infamy_check = false,
		remove_type = 1,
		Infamy_check_10 = 1,
		Infamy_check_15 = 1,
		Infamy_check_20 = 1,
		Infamy_check_24 = 1,
		Infamy_check_25 = 1,
		achievements_check = false,
		detection_loud = false,
		detection_loud_value = 15,
		detection_stealth = false,
		detection_stealth_value = 1,
		ammo_bag = false,
		doctor_bag = false,
		trip_mine = false,
		first_aid_kit = false,
		}
	NoobJoin.Name = "NGBTO"
	NoobJoin.Prefix = "[" .. NoobJoin.Name .. "]"
	NoobJoin.Update_message = {
		"Changelog:",
		"Updated at 01-09-2017",
		"BLT 2 SupportBLT 2 Support",
		"Fixed error print due to json files",
	}
	NoobJoin.Needed_message = 76
	NoobJoin.Players = {}
	local num_player_slots = BigLobbyGlobals and BigLobbyGlobals:num_player_slots() or 4
	for i=1,num_player_slots do -- peer info
		NoobJoin.Players[i] = {}
		for j=1,9 do -- Skills printed, pd2stats, skills overlay, join time, playtime, achievements, detection, cheater,
			NoobJoin.Players[i][j] = 0
		end
	end
	NoobJoin.Colors = {"ff0000", "00ff04", "1a64f6"} -- RGB
	NoobJoin.blacklist = {}
	NoobJoin.synced = {}
	 NoobJoin.Kick_on_join = {}
	NoobJoin.Blacklist_name = "blacklist.ini"
	-- Populate blacklist with steam ids
	-- old blacklist
	local file, err = io.open(NoobJoin._path .. NoobJoin.Blacklist_name, "r")
	if file then
		for line in file:lines() do
			if string.sub(line, 1,1) ~= ";" then
				table.insert(NoobJoin.blacklist, line)
			end
		end
	file:close()
	end
	-- new blacklist
	local file, err = io.open(NoobJoin.Blacklist_name, "r")
	if file then
		for line in file:lines() do
			if string.sub(line, 1,1) ~= ";" then
				table.insert(NoobJoin.blacklist, line)
			end
		end
	end
	if not file then
		file = io.open(NoobJoin.Blacklist_name, "w")
	end
	file:close()
	-- dr_newbie's blacklist
	local file1, err = io.open("kicklist.ini", "r")
	if file1 then
		for line in file1:lines() do
			table.insert(NoobJoin.blacklist, line)
		end
		file1:close()
	end
end

function NoobJoin:Load()
	local file = io.open(NoobJoin._data_path, "r")
	if file then
		for k, v in pairs(json.decode(file:read("*all")) or {}) do
			NoobJoin.settings[k] = v
		end
		file:close()
	end
end

function NoobJoin:Save()
	local file = io.open(NoobJoin._data_path, "w+")
	if file then
		file:write(json.encode(NoobJoin.settings))
		file:close()
	end
end

-- Load selected localization
local mpath=ModPath .. "loc/"
Hooks:Add("LocalizationManagerPostInit", "LocalizationManagerPostInit_NoobJoinInitial", function(loc)
	loc:load_localization_file(mpath.."initial.json")
	loc:load_localization_file(mpath.."english.json")
end)
Hooks:Add("LocalizationManagerPostInit", "LocalizationManagerPostInit_NoobJoin", function(loc)
	NoobJoin:Load()
	local path
	if NoobJoin.settings.Noob_language == 1 then
		path="english"
	elseif NoobJoin.settings.Noob_language == 2 then
		path="french"
	elseif NoobJoin.settings.Noob_language == 3 then
		path="russian"
	elseif NoobJoin.settings.Noob_language == 4 then
		path="german"
	elseif NoobJoin.settings.Noob_language == 5 then
		path="italian"
	elseif NoobJoin.settings.Noob_language == 6 then
		path="dutch"
	elseif NoobJoin.settings.Noob_language == 7 then
		path="spanish"
	elseif NoobJoin.settings.Noob_language == 8 then
		path="turkish"
	elseif NoobJoin.settings.Noob_language == 9 then
		path="chinese"
	elseif NoobJoin.settings.Noob_language == 10 then
		path="czech"
	elseif NoobJoin.settings.Noob_language == 11 then
		path="thai"
	end
	loc:load_localization_file(mpath..path .. ".json")
end)

-- Menu create
Hooks:Add("MenuManagerInitialize", "MenuManagerInitialize_NoobJoin", function(menu_manager)
	MenuCallbackHandler.Toggle = function(self)
		MenuCallbackHandler:Self_Inspect()
	end

	MenuCallbackHandler.SkillInfo = function(self)
		MenuCallbackHandler:Skill_Show()
	end

	MenuCallbackHandler.ManualBan = function(self)
		MenuCallbackHandler:Manual_banning()
	end

	MenuCallbackHandler.PlayerInspect = function(self)
		MenuCallbackHandler:Player_inspection_menu()
	end

	--Main menu
	MenuCallbackHandler.ngbto_language_callback = function(this, item)
		NoobJoin.settings.Noob_language = item:value()
	end

	--Information
	MenuCallbackHandler.ngbto_broadcast_callback = function(this, item)
		NoobJoin.settings.broadcastinfo_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_show_skills_callback = function(this, item)
		NoobJoin.settings.show_perks_info_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_detailed_skills_callback = function(this, item)
		NoobJoin.settings.skills_detailed = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_statistics_callback = function(this, item)
		NoobJoin.settings.Stats_Print = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_remove_type_callback = function(this, item)
		NoobJoin.settings.remove_type = item:value()
	end
	MenuCallbackHandler.ngbto_kick_type_callback = function(this, item)
		NoobJoin.settings.kick_type = Utils:ToggleItemToBoolean(item)
	end

	-- Anti-cheat
	MenuCallbackHandler.ngbto_pd2stats_callback = function(this, item)
		NoobJoin.settings.usepd2statsanticheat = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_pd2stats_extra_callback = function(this, item)
		NoobJoin.settings.tag_not_enough_heists_completed = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_skill_cheats_callback = function(this, item)
		NoobJoin.settings.skill_cheats = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_ingame_anticheat_callback = function(this, item)
		NoobJoin.settings.ingame_anticheat = Utils:ToggleItemToBoolean(item)
	end

	--Hour Filters
	
	MenuCallbackHandler.ngbto_one_down_loud_callback = function(this, item)
		NoobJoin.settings.min_hours_loud_sm_wish_val = item:value()
	end
	MenuCallbackHandler.ngbto_one_down_stealth_callback = function(this, item)
		NoobJoin.settings.min_hours_stealth_sm_wish_val = item:value()
	end
	MenuCallbackHandler.ngbto_mayhem_loud_callback = function(this, item)
		NoobJoin.settings.min_hours_loud_easy_wish_val = item:value()
	end
	MenuCallbackHandler.ngbto_mayhem_stealth_callback = function(this, item)
		NoobJoin.settings.min_hours_stealth_easy_wish_val = item:value()
	end
	
	MenuCallbackHandler.ngbto_deathwish_loud_callback = function(this, item)
		NoobJoin.settings.min_hours_loud_dw_val = item:value()
	end
	MenuCallbackHandler.ngbto_deathwish_stealth_callback = function(this, item)
		NoobJoin.settings.min_hours_stealth_dw_val = item:value()
	end
	MenuCallbackHandler.ngbto_overkill_loud_callback = function(this, item)
		NoobJoin.settings.min_hours_loud_ovk_val = item:value()
	end
	MenuCallbackHandler.ngbto_overkill_stealth_callback = function(this, item)
		NoobJoin.settings.min_hours_stealth_ovk_val = item:value()
	end
	MenuCallbackHandler.ngbto_total_heists_callback = function(this, item)
		NoobJoin.settings.total_count_enable = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_deathwish_heists_callback = function(this, item)
		NoobJoin.settings.deathwish_count_enable = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_total_heists_number_callback = function(this, item)
		NoobJoin.settings.total_count = item:value()
	end
	MenuCallbackHandler.ngbto_deathwish_heists_number_callback = function(this, item)
		NoobJoin.settings.deathwish_count = item:value()
	end

	--Hidden filters
	MenuCallbackHandler.ngbto_hidden_master_callback = function(this, item)
		NoobJoin.settings.kickhidden_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_hidden_infamy_callback = function(this, item)
		NoobJoin.settings.hiddeninfamycheck = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_hidden_infamy_minimum_callback = function(this, item)
		NoobJoin.settings.hiddeninfamy = item:value()
	end
	MenuCallbackHandler.ngbto_hidden_infamy_maximum_callback = function(this, item)
		NoobJoin.settings.hiddeninfamymax = item:value()
	end
	MenuCallbackHandler.ngbto_hidden_infamy_v_callback = function(this, item)
		NoobJoin.settings.hiddenv = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_hidden_infamy_xxv_callback = function(this, item)
		NoobJoin.settings.hiddenxxv = Utils:ToggleItemToBoolean(item)
	end

	--Perk decks
	MenuCallbackHandler.ngbto_incomplete_perk_deck_callback = function(this, item)
		NoobJoin.settings.incompletedeck = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_empty_perk_deck_callback = function(this, item)
		NoobJoin.settings.perkdeck0_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_1_callback = function(this, item)
		NoobJoin.settings.perkdeck_1_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_2_callback = function(this, item)
		NoobJoin.settings.perkdeck_2_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_3_callback = function(this, item)
		NoobJoin.settings.perkdeck_3_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_4_callback = function(this, item)
		NoobJoin.settings.perkdeck_4_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_5_callback = function(this, item)
		NoobJoin.settings.perkdeck_5_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_6_callback = function(this, item)
		NoobJoin.settings.perkdeck_6_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_7_callback = function(this, item)
		NoobJoin.settings.perkdeck_7_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_8_callback = function(this, item)
		NoobJoin.settings.perkdeck_8_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_9_callback = function(this, item)
		NoobJoin.settings.perkdeck_9_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_10_callback = function(this, item)
		NoobJoin.settings.perkdeck_10_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_11_callback = function(this, item)
		NoobJoin.settings.perkdeck_11_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_12_callback = function(this, item)
		NoobJoin.settings.perkdeck_12_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_13_callback = function(this, item)
		NoobJoin.settings.perkdeck_13_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_14_callback = function(this, item)
		NoobJoin.settings.perkdeck_14_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_15_callback = function(this, item)
		NoobJoin.settings.perkdeck_15_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_16_callback = function(this, item)
		NoobJoin.settings.perkdeck_16_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_17_callback = function(this, item)
		NoobJoin.settings.perkdeck_17_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_18_callback = function(this, item)
		NoobJoin.settings.perkdeck_18_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_19_callback = function(this, item)
		NoobJoin.settings.perkdeck_19_val = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_perk_deck_20_callback = function(this, item)
		NoobJoin.settings.perkdeck_20_val = Utils:ToggleItemToBoolean(item)
	end
	
	--Deployables filters
	MenuCallbackHandler.ngbto_no_deployable_callback = function(this, item)
		NoobJoin.settings.noequipment = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_four_first_aid_kits_callback = function(this, item)
		NoobJoin.settings.fourfaks = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_single_turret_callback = function(this, item)
		NoobJoin.settings.singleturret = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_body_bags_callback = function(this, item)
		NoobJoin.settings.bodybags = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_armor_kit_callback = function(this, item)
		NoobJoin.settings.armorkit = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_single_ecm_callback = function(this, item)
		NoobJoin.settings.singleecm = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_ammo_bag_callback = function(this, item)
		NoobJoin.settings.ammo_bag = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_doctor_bag_callback = function(this, item)
		NoobJoin.settings.doctor_bag = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_trip_mine_callback = function(this, item)
		NoobJoin.settings.trip_mine = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_first_aid_kit_callback = function(this, item)
		NoobJoin.settings.first_aid_kit = Utils:ToggleItemToBoolean(item)
	end

	--Armor
	MenuCallbackHandler.ngbto_armor_1_callback = function(this, item)
		NoobJoin.settings.armor1 = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_armor_2_callback = function(this, item)
		NoobJoin.settings.armor2 = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_armor_3_callback = function(this, item)
		NoobJoin.settings.armor3 = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_armor_4_callback = function(this, item)
		NoobJoin.settings.armor4 = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_armor_5_callback = function(this, item)
		NoobJoin.settings.armor5 = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_armor_6_callback = function(this, item)
		NoobJoin.settings.armor6 = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_armor_7_callback = function(this, item)
		NoobJoin.settings.armor7 = Utils:ToggleItemToBoolean(item)
	end

	--Misc
	MenuCallbackHandler.ngbto_detection_loud_callback = function(this, item)
		NoobJoin.settings.detection_loud = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_detection_loud_value_callback = function(this, item)
		NoobJoin.settings.detection_loud_value = item:value()
	end
	MenuCallbackHandler.ngbto_detection_stealth_callback = function(this, item)
		NoobJoin.settings.detection_stealth = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.ngbto_detection_stealth_value_callback = function(this, item)
		NoobJoin.settings.detection_stealth_value = item:value()
	end
	MenuCallbackHandler.achievements_check_callback = function(this, item)
		NoobJoin.settings.achievements_check = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.Infamy_check_callback = function(this, item)
		NoobJoin.settings.Infamy_check = Utils:ToggleItemToBoolean(item)
	end
	MenuCallbackHandler.Infamy_check_10_callback = function(this, item)
		NoobJoin.settings.Infamy_check_10 = item:value()
	end
	MenuCallbackHandler.Infamy_check_15_callback = function(this, item)
		NoobJoin.settings.Infamy_check_15 = item:value()
	end
	MenuCallbackHandler.Infamy_check_20_callback = function(this, item)
		NoobJoin.settings.Infamy_check_20 = item:value()
	end
	MenuCallbackHandler.Infamy_check_24_callback = function(this, item)
		NoobJoin.settings.Infamy_check_24 = item:value()
	end
	MenuCallbackHandler.Infamy_check_25_callback = function(this, item)
		NoobJoin.settings.Infamy_check_25 = item:value()
	end
	MenuCallbackHandler.ngbto_menu = function(this, item)
		NoobJoin:Save()
	end
	NoobJoin:Load()
	--Submenus
	MenuHelper:LoadFromJsonFile(NoobJoin._path .. "menu/main.json", NoobJoin, NoobJoin.settings)
	MenuHelper:LoadFromJsonFile(NoobJoin._path .. "menu/extras.json", NoobJoin, NoobJoin.settings)
	MenuHelper:LoadFromJsonFile(NoobJoin._path .. "menu/anti_cheat.json", NoobJoin, NoobJoin.settings)
	MenuHelper:LoadFromJsonFile(NoobJoin._path .. "menu/data_filters.json", NoobJoin, NoobJoin.settings)
	MenuHelper:LoadFromJsonFile(NoobJoin._path .. "menu/hidden.json", NoobJoin, NoobJoin.settings)
	MenuHelper:LoadFromJsonFile(NoobJoin._path .. "menu/perk_decks.json", NoobJoin, NoobJoin.settings)
	MenuHelper:LoadFromJsonFile(NoobJoin._path .. "menu/deployables.json", NoobJoin, NoobJoin.settings)
	MenuHelper:LoadFromJsonFile(NoobJoin._path .. "menu/armor.json", NoobJoin, NoobJoin.settings)
	MenuHelper:LoadFromJsonFile(NoobJoin._path .. "menu/misc.json", NoobJoin, NoobJoin.settings)
end)
